rust/src/measurements/make_private_expr/expr_laplace/mod.rs

```diff
@@ -0,0 +1,320 @@
+        // DataType::UInt16 => laplace_impl_integer::<UInt16Type>(series, scale),
+        DataType::UInt32 => laplace_impl_integer::<UInt32Type>(series, scale),
+        DataType::UInt64 => laplace_impl_integer::<UInt64Type>(series, scale),
+        DataType::Int8 => laplace_impl_integer::<Int8Type>(series, scale),
+        DataType::Int16 => laplace_impl_integer::<Int16Type>(series, scale),
+        DataType::Int32 => laplace_impl_integer::<Int32Type>(series, scale),
+        DataType::Int64 => laplace_impl_integer::<Int64Type>(series, scale),
+        DataType::Float32 => laplace_impl_float::<Float32Type>(series, scale),
+        DataType::Float64 => laplace_impl_float::<Float64Type>(series, scale),
+        _ => Err(PolarsError::ComputeError(
+            "Expected numeric data type.".into(),
+        )),
+    }
+}
+
+/// Helper function for the Polars plan optimizer to determine the output type of the expression.
+///
+/// Ensures that the input field is numeric.
+pub(crate) fn laplace_type_udf(input_fields: &[Field]) -> PolarsResult<Field> {
+    let [field] = <&[Field; 1]>::try_from(input_fields).map_err(|_| {
+        PolarsError::InvalidOperation("Identity expects a single input field".into())
+    })?;
+    use DataType::*;
+    if !matches!(
+        field.data_type(),
+        UInt8 | UInt16 | UInt32 | UInt64 | Int8 | Int16 | Int32 | Int64 | Float32 | Float64
+    ) {
+        return Err(polars_err!(
+            InvalidOperation: "Expected numeric data type, found {:?}",
+            field.data_type()
+        ));
+    }
+    Ok(field.clone())
+}
+
+// generate the FFI plugin for the Laplace noise expression
+#[cfg(feature = "ffi")]
+#[polars_expr(output_type_func=laplace_type_udf)]
+fn laplace(inputs: &[Series], kwargs: LaplaceArgs) -> PolarsResult<Series> {
+    laplace_udf(inputs, kwargs)
+}
+
+#[cfg(test)]
+mod test_make_laplace_expr {
+    use super::*;
+    use polars::prelude::*;
+
+    use crate::{
+        core::PrivacyNamespaceHelper,
+        measurements::PrivateExpr,
+        metrics::{InsertDeleteDistance, PartitionDistance, SymmetricDistance},
+        transformations::polars_test::get_test_data,
+    };
+
+    #[test]
+    fn test_make_laplace_expr() -> Fallible<()> {
+        let (lf_domain, lf) = get_test_data()?;
+        let expr_domain = lf_domain.select();
+        let scale: f64 = 0.0;
+
+        let m_lap = col("B")
+            .clip(lit(1.), lit(2.))
+            .sum()
+            .dp()
+            .laplace(f64::NAN)
+            .make_private(
+                expr_domain,
+                SymmetricDistance,
+                MaxDivergence::default(),
+                scale,
+            )?;
+
+        let meas_res = m_lap.invoke(&(lf.logical_plan.clone(), all()))?;
+
+        let df_act = lf.clone().select([meas_res]).collect()?;
+        let df_exp = lf.clone().select([sum("B")]).collect()?;
+
+        assert_eq!(df_act, df_exp);
```

Comment:
So I guess we're not actually comparing DP results here, so I'm confused about what this test is doing.

Reply:
With zero scale, the test becomes deterministic.